contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -596,10 +607,42 @@ add_query_hints(PLtsql_expr *expr)
 	if (!leading_hint.empty())
 		hint += leading_hint;
 	hint += "*/";
+	return hint;
+}
+
+static char *
+format_query_with_hints(char *query, std::string hints)
+{
 	StringInfoData new_query;
 	initStringInfo(&new_query);
-	appendStringInfo(&new_query, "%s %s", const_cast <char *>(hint.c_str()), expr->query);
-	expr->query = new_query.data;
+	int initialTokenEnd = find_hint_offset(query);
+
+	appendStringInfo(&new_query, "%s", psprintf("%.*s", initialTokenEnd, query));
+	appendStringInfo(&new_query, " %s", const_cast <char *>(hints.c_str()));
+	appendStringInfo(&new_query, " %s", &query[initialTokenEnd]);
+	return new_query.data;
+}
+
+static int
+find_hint_offset(char *query)
+{
+	std::string queryString(query);
+	size_t spaceIdx = queryString.find_first_of(" \t\r\n");
```

Comment:
Could you check whether we can use `\s`?

Reply:
\s is not supported it seems, but I updated to include all space characters defined here https://cplusplus.com/reference/cctype/isspace/